yt/frontends/cf_radial/data_structures.py

```diff
@@ -0,0 +1,269 @@
+"""
+CF Radial data structures
+
+
+
+"""
+
+import os
+import stat
+import weakref
+
+import numpy as np
+
+from yt.data_objects.index_subobjects.grid_patch import AMRGridPatch
+from yt.data_objects.static_output import Dataset
+from yt.funcs import mylog
+from yt.geometry.grid_geometry_handler import GridIndex
+from yt.utilities.on_demand_imports import _xarray as xr
+
+from .fields import CFRadialFieldInfo
+
+
+class CFRadialGrid(AMRGridPatch):
+    _id_offset = 0
+
+    def __init__(self, id, index, level, dimensions):
+        super().__init__(id, filename=index.index_filename, index=index)
+        self.Parent = None
+        self.Children = []
+        self.Level = level
+        self.ActiveDimensions = dimensions
+
+    def __repr__(self):
+        return "CFRadialGrid_%04i (%s)" % (self.id, self.ActiveDimensions)
+
+
+class CFRadialHierarchy(GridIndex):
+    grid = CFRadialGrid
+
+    def __init__(self, ds, dataset_type="cf_radial"):
+        self.dataset_type = dataset_type
+        self.dataset = weakref.proxy(ds)
+        # for now, the index file is the dataset!
```

Comment:
this was likely a long term decision: I suppose if the data were split up across netcdf files on their own grids this might change? But that seems unlikely and would require refactoring large portions of this frontend... So I adjusted this comment.